### ISSUE TITLE
Loss function

### ISSUE THREAD

**Issue by yerim21c** [yerim21c/Deeplearning#6]
# 손실 함수(Loss function)

## 1. 오차제곱합(SSE)

가장 많이 쓰이는 손실 함수로 y_k는 신경망의 출력, t_k는 정답 레이블, k는 데이터의 차원 수를 나타낸다.
![image](https://user-images.githubusercontent.com/87102353/126345721-df89eee7-95ce-4b52-b4c8-dba97becebde.png)

~~~ python
import numpy as np

def sum_squares_error(y, t):
    return 0.5 * np.sum((y-t)**2)

t = [0, 0, 1, 0, 0, 0, 0, 0, 0, 0]

# 2일 확률이 가장 높다고 추정한 경우
y = [0.1, 0.05, 0.6, 0.0, 0.05, 0.1, 0.0, 0.1, 0.0, 0.0]
sum_squares_error(np.array(y), np.array(t))
# 0.09750000000000003 -> 손실 함수의 추정 결과가 더 작으니 정답이 더 가까울 것으로 판단

# 7일 확률이 가장 높다고 추정한 경우
y = [0.1, 0.05, 0.1, 0.0, 0.05, 0.1, 0.0, 0.6, 0.0, 0.0]
sum_squares_error(np.array(y), np.array(t))
# 0.5975 
~~~

## 2. 교차 엔트로피 오차(Cross entropy error)
교차 엔트로피 오차에서 정답에 해당하는 출력이 커질수록 0에 다가가다가, 그 출력이 1일 때 0이 된다. 반대로 정답일 때의 출력이 작아질수록 오차는 커진다. 
![image](https://user-images.githubusercontent.com/87102353/126348278-fdf12e9e-ce43-4238-a4fa-0767a5e6e1f3.png)

~~~ python
import numpy as np
def cross_entropy_error(y, t):
    delta = 1e-7
    return -np.sum(t * np.log(y + delta))  # np.log() 함수에 0을 입력하면 -inf가 되어 계산을 진행할 수 없기 때문에 delta 값을 더함

t = [0, 0, 1, 0, 0, 0, 0, 0, 0, 0]

# 2일 확률이 가장 높다고 추정한 경우
y = [0.1, 0.05, 0.6, 0.0, 0.05, 0.1, 0.0, 0.1, 0.0, 0.0]
cross_entropy_error(np.array(y), np.array(t))
# 0.510825457099338

# 7일 확률이 가장 높다고 추정한 경우
y = [0.1, 0.05, 0.1, 0.0, 0.05, 0.1, 0.0, 0.6, 0.0, 0.0]
cross_entropy_error(np.array(y), np.array(t))
# 2.302584092994546
~~~
